src/service_control_client_impl.cc

```diff
@@ -62,19 +82,30 @@ ServiceControlClientImpl::ServiceControlClientImpl(
     // Class members cannot be captured in lambda. We need to make a copy to
     // support C++11.
     std::shared_ptr<CheckAggregator> check_aggregator_copy = check_aggregator_;
+    std::shared_ptr<QuotaAggregator> quota_aggregator_copy = quota_aggregator_;
     std::shared_ptr<ReportAggregator> report_aggregator_copy =
         report_aggregator_;
+
     flush_timer_ = options.periodic_timer(
-        flush_interval, [check_aggregator_copy, report_aggregator_copy]() {
+        flush_interval, [check_aggregator_copy, quota_aggregator_copy,
+                         report_aggregator_copy]() {
+
           Status status = check_aggregator_copy->Flush();
           if (!status.ok()) {
-            GOOGLE_LOG(ERROR) << "Failed in Check::Flush() "
+            GOOGLE_LOG(ERROR)
+                << "Failed in Check::Flush() " << status.error_message();
+          }
+
+          status = quota_aggregator_copy->Flush();
```

Comment:
quota_aggregator_copy->Flush() is calling following code. It removes expired entries only.

  if (cache_) {
    cache_->RemoveExpiredEntries();
  }